src/components/side_panel/remove_duplicates/remove_duplicates.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, onWillUpdateProps, useState } from "@odoo/owl";
+import { numberToLetters, zoneToDimension } from "../../../helpers";
+import { _t } from "../../../translation";
+import { HeaderIndex, SpreadsheetChildEnv } from "../../../types/index";
+import { css } from "../../helpers";
+import { RemoveDuplicateTerms } from "../../translations_terms";
+import { SidePanelErrors } from "../side_panel_errors/side_panel_errors";
+
+css/* scss */ `
+  .o-checkbox-selection {
+    height: 150px;
+  }
+`;
+
+interface Props {
+  onCloseSidePanel: () => void;
+}
+
+interface RemoveDuplicatesState {
+  hasHeader: boolean;
+  columns: { [colIndex: number]: boolean };
+}
+
+export class RemoveDuplicatesPanel extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-RemoveDuplicatesPanel";
+  static components = { SidePanelErrors };
+
+  state: RemoveDuplicatesState = useState({
+    hasHeader: false,
+    columns: {},
+  });
+
+  setup() {
+    onWillUpdateProps(() => this.updateColumns());
+  }
+
+  toggleHasHeader() {
+    this.state.hasHeader = !this.state.hasHeader;
+  }
+
+  toggleAllColumns() {
+    const newState = !this.isEveryColumnSelected;
+    for (let index in this.state.columns) {
+      this.state.columns[index] = newState;
+    }
+  }
+
+  toggleColumn(colIndex: number) {
+    this.state.columns[colIndex] = !this.state.columns[colIndex];
+  }
+
+  onRemoveDuplicates() {
+    this.env.model.dispatch("REMOVE_DUPLICATES", {
+      hasHeader: this.state.hasHeader,
+      columns: this.getColsToAnalyze(),
+    });
+  }
+
+  getColLabel(colKey: string): string {
+    const col = parseInt(colKey);
+    let colLabel = _t("Column %s", numberToLetters(col));
```

Comment:
if the question is "why can't we use colKey in numberToLetters":

getColLabel is used in the template and will receive the colIndex from "Object.keys(state.columns)". The problem with Object.keys is that it turns keys into strings when they are numbers.
And if we pass a string to "numberToLetters" it returns an incorrect result...

maybe if we use array.from on state.columns.keys() (if columns is a map) --> it could solve the problem


:)